### ISSUE TITLE
re-attach to session

### ISSUE THREAD

**Issue by ryn1x** [MisterTea/EternalTerminal#365]
How can I re-attach to a session if my local shell/computer crashes? When I use et again I get a new shell on the remote machine. A quick `w` shows my other session and job is still running.

I also noticed looking through previous issues before posting that only 1 et connection can run at a time (issue from 2017), but I have been running multiple. Maybe this has changed since that old issue.

**Comment by MisterTea** [comment]
Reattach would require spilling the et state to disk & a rewind capability, which aren't implemented yet.

**Comment by Xtricman** [comment]
Why would it require that? When a new connection is established, etserver can simply start to sync the state of an choosen etterminal to the client side, just as mosh did. When the old connection crashed, the etterminal is still running.  

Every client can query etterminals on the server and tell etserver which etterminal it wanna sync with. The new connection don't need to use the original session key of the old connection, simply generate another key.

> Reattach would require spilling the et state to disk & a rewind capability

**Comment by MisterTea** [comment]
@jwmcglynn what are your thoughts?

**Comment by ryuheechul** [comment]
I'm also "surprised" by the lack of "auto reattaching" behavior. (not that there is any indication that ET should work that way, but that was my natural expectation for a tool like ET/mosh)

I intentionally severed a connection by killing a pane inside tmux and ran `et [my-server]` again and it just started a new shell.
But I can also list the process that I was running (neovim) was still running as a dependent process of `etterminal`.

These two facts don't give me a clear path forward after re-connection:
- previous session appears seemingly gone (visually when manually reconnecting)
- but it's actually tucked away and alive and hogging up the resources until manually track it down and clean up

Am I expected to manually track down and clean up?

When I tested this behavior with mosh, I'm also surprised with these:
- mosh also doesn't reconnect to a "severed connection" (again by killing a tmux panel) just like ET (I thought mosh would as I initially thought so for ET)
- but there is no dangling process left

Now I'm moving on to sharing my opinion.

What I would like to see is a prompt (if not automatic rejoining with a message)  asking a) if I would like to attach to a "dangling" session if there is any like in case of the nvim process under zsh under `etterminal` or b) simply to stop the dangling one.

I came with this perspective just because how I particularly set up my server's shell. I use Zellij to create or attach to a session (inside the remote host) named "ssh" with `zlogin` file (only in case it's via SSH connection) so an accidental disconnection (e.g. by a crash on client side) still preserve the session and allow me to be in the same screen as soon as I reconnect. So I can understand this is not necessarily the behavior that majority of people expect but just wanted to share my perspective on this issue.

Now I can kind of see the main concern that ET solves is the fragile connection while when (at least) the client ET process is still running which is great but if ET also can expand the definition of disruption that transcends existence of the client process in some way, that would be quite nice.

**Comment by heyakyra** [comment]
I agree this would need some history of recent connections to attempt to revive the latest one (or a subcommand to do so) rather than the default of a new connection. `reptyr` is very useful, but it's not the most usable to have to look up the `pid` with [`btm`](https://bottom.pages.dev/stable/), [`htop`](https://htop.dev/), [`gtop`](https://github.com/aksakalli/gtop), [`gotop`](https://github.com/xxxserxxx/gotop), or old-school `top`. 

This is also relevant to multiplexed terminal session sharing tools like [`zmate`](https://github.com/ziinaio/zmate/issues/19), [devsession](https://github.com/devsession-org/devsession-org.github.io/issues/1), [`upterm`](https://github.com/owenthereal/upterm/issues/510), and [`tmate`](https://github.com/tmate-io/tmate/issues/330)
